examples/deepwalk/model.py

```diff
@@ -28,53 +28,85 @@ def __init__(self,
-
-            self.emb = emb_func
+            self.u_emb = u_emb_func
+            if self.shared_embedding:
+                def v_emb_func(x):
+                    d_shape = paddle.shape(x)
+                    x_emb = paddle.static.nn.sparse_embedding(
+                        paddle.reshape(x, [-1, 1]), [num_nodes, embed_size],
+                        param_attr=v_emb_attr)
+                    return paddle.reshape(x_emb,
+                                          [d_shape[0], d_shape[1], embed_size])
+                self.v_emb = v_emb_func
         elif num_emb_part > 1:
             assert embed_size % num_emb_part == 0
-            emb_list = []
+            u_emb_list = []
             for i in range(num_emb_part):
-                emb_attr = paddle.ParamAttr(
-                    name="node_embedding_part%s" % i, initializer=embed_init)
-                emb = nn.Embedding(
+                u_emb_attr = paddle.ParamAttr(
+                    name="u_node_embedding_part%s" % i, initializer=u_embed_init)
+                u_emb = nn.Embedding(
                     num_nodes,
                     embed_size // num_emb_part,
-                    weight_attr=emb_attr)
-                emb_list.append(emb)
-            self.emb_list = nn.LayerList(emb_list)
-            self.emb = lambda x: paddle.concat([emb(x) for emb in emb_list], -1)
+                    weight_attr=u_emb_attr)
+                u_emb_list.append(u_emb)
+            self.u_emb_list = nn.LayerList(u_emb_list)
+            self.u_emb = lambda x: paddle.concat([emb(x) for emb in u_emb_list], -1)
+            if self.shared_embedding:
+                v_emb_list = []
+                for i in range(num_emb_part):
+                    v_emb_attr = paddle.ParamAttr(
+                        name="v_node_embedding_part%s" % i, initializer=v_embed_init)
+                    v_emb = nn.Embedding(
+                        num_nodes,
+                        embed_size // num_emb_part,
+                        weight_attr=v_emb_attr)
+                    v_emb_list.append(v_emb)
+                self.v_emb_list = nn.LayerList(v_emb_list)
+                self.v_emb = lambda x: paddle.concat([emb(x) for emb in v_emb_list], -1)
         else:
-            self.emb = nn.Embedding(
-                num_nodes, embed_size, sparse=sparse, weight_attr=emb_attr)
+            self.u_emb = nn.Embedding(
+                num_nodes, embed_size, sparse=sparse, weight_attr=u_emb_attr)
+            if self.shared_embedding:
+                self.v_emb = nn.Embedding(
+                    num_nodes, embed_size, sparse=sparse, weight_attr=v_emb_attr)
         self.loss = paddle.nn.BCEWithLogitsLoss()
 
     def forward(self, src, dsts):
         # src [b, 1]
         # dsts [b, 1+neg]
 
-        src_embed = self.emb(src)
-        dsts_embed = self.emb(dsts)
+        src_embed = self.u_emb(src)
+        if self.shared_embedding:
```

Comment:
The argument shared_embedding setting of initialization and  forward are inconsistent. Please run the code for test.

Reply:
Okay, a stupid mistake...